projects/js-toolkit/packages/liferay-cli/src/new/util/prompt.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import {format} from '@liferay/js-toolkit-core';
+import inquirer from 'inquirer';
+
+import type {OptionValue, Options} from '../index';
+
+interface DefaultFunction {
+	(options: Options): OptionValue;
+}
+
+interface ReduceablePrompt {
+	default: DefaultFunction | OptionValue;
+	message: string;
+	name: string;
+}
+
+const {print, question} = format;
+
+export default async function prompt(
+	options: Options,
+	prompts: inquirer.QuestionCollection
+): Promise<Options> {
+	if (options.batch) {
+		return (prompts as ReduceablePrompt[]).reduce((options, prompt) => {
+			let answer = prompt.default;
+
+			if (typeof answer === 'function') {
+				answer = answer(options);
+			}
+
+			print(question`${prompt.message} {${answer}}`);
+
+			options[prompt.name] = answer;
+
+			return options;
+		}, options);
```

Comment:
I don't think you actually need a `reduce` here since you are just modifying the original object passed in.

```suggestion
		return (prompts as ReduceablePrompt[]).forEach((prompt) => {
			let answer = prompt.default;

			if (typeof answer === 'function') {
				answer = answer(options);
			}

			print(question`${prompt.message} {${answer}}`);

			options[prompt.name] = answer;
		});
```

Reply:
Not really, because you need to return the options object and [forEach doesn't return anything](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#return_value).

However, we can turn the `prompt` function into an in-place modification function. So, instead of:

```javascript
function prompt(options: Options) : Options
```

Make it:

```javascript 
function prompt(options: Options): void
```

Though I really prefer the first option, because it looks more natural to receive the current options and return the new ones.

And if we want to be more FP, I suppose we can introduce this change in the code:

```javascript
return (prompts /*...*/).reduce((options, prompt) => { /*...*/ }, {...options});
```

To make it explicit that we are not transforming the input options in any way.